### PR TITLE
fix: s3compatible endpoint validation

### PR DESCRIPTION
**Pull request by stehessel**
### Description

I thought to change the endpoint validation slightly to be more in line with other features. Using `FormatURL` avoid manual formatting and covers more edge cases. It also allows for `http` endpoints if they are explicitly defined as such (may occur when users are self-hosting their providers). I also changed the help text in the UI to account for `http` being allowed.

### User-facing documentation

(*must be* 2 items and both *must be* checked)

- [ ] CHANGELOG is updated
- [x] CHANGELOG update is not needed
- [ ] Documentation PR is created and linked above
- [x] Documentation is not needed

### Testing

- [x] inspected CI results

#### Automated testing

(*must be* at least 1 item and all items *must be* checked)

- [x] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [ ] modified existing tests
- [ ] contributed **no automated tests**
  

#### How I validated my change

qa-e2e already test different endpoint schemes.

### CODE REVIEW

File: central/externalbackups/plugins/s3compatible/s3compatible_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+			sanitizedEndpoint: "https://play.min.io",
+		},
+		"with http prefix": {
+			endpoint:          "http://play.min.io",
+			sanitizedEndpoint: "http://play.min.io",
+		},
+		"without prefix": {
+			endpoint:          "play.min.io",
+			sanitizedEndpoint: "https://play.min.io",
+		},
```
Comment (lvalerom) [inline central/externalbackups/plugins/s3compatible/s3compatible_test.go:27]: [nit] Maybe add a test case that tests the removal of trailing slash:
```suggestion
		"without prefix": {
			endpoint:          "play.min.io",
			sanitizedEndpoint: "https://play.min.io",
		},
		"with trailing slash": {
			endpoint:          "https://play.min.io/",
			sanitizedEndpoint: "https://play.min.io",
		},
```

**Review by lvalerom — APPROVED** [review]
LGTM. We will have to update the docs too because I think we wrote the endpoint needed to be `https`.

File: central/externalbackups/plugins/s3compatible/s3compatible_test.go (hunk 1 above)
Comment (stehessel) [reply central/externalbackups/plugins/s3compatible/s3compatible_test.go:27]: good idea, done

**Pull request merged**